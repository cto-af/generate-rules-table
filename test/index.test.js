import {createRulesTable} from '../lib/index.js';
import {fileURLToPath} from 'node:url';
import fs from 'node:fs/promises';
import path from 'node:path';
import {rejects} from 'node:assert';
import snap from 'snappy-snaps';
// eslint-disable-next-line n/no-unsupported-features/node-builtins
import test from 'node:test';

const fixtures = fileURLToPath(new URL('./fixtures/', import.meta.url));
const good = path.resolve(fixtures, 'good');
const bad = path.resolve(fixtures, 'bad');

test('createRulesTable', async() => {
  await createRulesTable({cwd: good});
  await snap('good', await fs.readFile(path.join(good, 'README.md'), 'utf8'));
});

test('bad README', async() => {
  await rejects(
    () => createRulesTable({cwd: bad}),
    {message: 'Invalid markdown file, needs start/end comments'}
  );
});

test('bad package.json', async() => {
  await rejects(
    () => createRulesTable({cwd: '/'}),
    {message: 'Unable to find package.json from "/"'}
  );
});
