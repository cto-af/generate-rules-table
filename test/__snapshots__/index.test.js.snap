// Data Snap v1

exports[`good`] = {
  "data": "Test generation:\n\n\u003C!-- Rule Table Start --\u003E\n| **Rule Name** | **Description** | **Recommended** |\n| :- | :- | :-: |\n| [`foo`](.\u002Fdocs\u002Frules\u002Ffoo.md) | Foo test | yes |\n| [`bar`](.\u002Fdocs\u002Frules\u002Fbar.md) | Bar test | no |\n\u003C!-- Rule Table End --\u003E\n\nWith text after.\n",
  "meta": {
    "expires": null
  }
}
